lib/perl/Genome/Model/Tools/Vcf/EvaluateVcf.pm

```diff
@@ -0,0 +1,790 @@
+    my $prg = $base->subdir('bin')->file('bedtools');
+    return $prg->stringify;
+}
+
+sub vcflib_tool {
+    my ($self, $tool) = @_;
+    my $path = Genome::Model::Tools::Vcflib->vcflib_tool_path(
+        $tool, $self->vcflib_version
+    );
+    return $path;
+}
+
+sub reference {
+    my $self = shift;
+    return "/gscmnt/ams1102/info/model_data/"
+           . "2869585698/build106942997/all_sequences.fa";
+}
+
+sub _process_input_file {
+    my ($self, $input_file, $output_file) = @_;
+    my @cmds = (
+        $self->restrict_commands($input_file, $self->roi),
+        $self->restrict_to_sample_commands("/dev/stdin", $self->old_sample),
+        $self->pass_only_commands("/dev/stdin", $self->pass_only_expression),
+        $self->allelic_primitives_commands("/dev/stdin"),
+        $self->normalize_vcf_commands("/dev/stdin", $self->reference),
+        $self->sort_commands("/dev/stdin"),
+        $self->restrict_commands("stdin", $self->roi),
+        "bgzip -c",
+        );
+
+    my $cmd = Genome::Sys::ShellPipeline->create(
+        pipe_commands => \@cmds,
+        redirects => " > $output_file",
+        );
+    $cmd->execute;
+
+}
+
+sub _clean_vcf {
+    my ($self, $input_file, $output_file) = @_;
+
+    $self->status_message("Cleaning vcf $input_file => $output_file");
+    $DB::single = 1;
+    my $reader = Genome::File::Vcf::Reader->new($input_file);
+    my $writer = Genome::File::Vcf::Writer->new($output_file, $reader->header);
+
+    $reader->add_filter(_make_bad_indel_filter($self->roi)) if $self->clean_indels;
+    $reader->add_filter(_make_per_allele_info_filter($reader->header));
+
+    while (my $entry = $reader->next) {
+        $writer->write($entry);
+    }
+    $writer->close;
+}
+
+sub _make_per_allele_info_filter {
+    my ($header) = @_;
+
+    my $info = $header->info_types;
+    my @remove = grep {$info->{$_}{number} =~ /^(A|R)$/} keys %$info;
+
+    return sub {
+        my $entry = shift;
+        my $info_hash = $entry->info;
+        delete @{$entry}{@remove};
+        return $entry;
+    };
+}
+
+sub _make_bad_indel_filter {
+    my ($roi_file) = @_;
+
+    my %bed_ends;
+
+    # This is compressed!
+    my $fh = IO::Zlib->new;
+    $fh->open($roi_file, "rb") or die "Unable to open BED file $roi_file for removal of bad indel lines\n";
+    while(my $bedline = $fh->getline) {
+        chomp $bedline;
+        my ($chr, $start, $stop) = split "\t", $bedline;
+        $bed_ends{"$chr\t$stop"} = 1;
+    }
+    $fh->close;
+
+    return sub {
+        my $entry = shift;
+        my $x = sprintf "%s\t%s", $entry->{chrom}, $entry->{position};
+        return if exists $bed_ends{$x};
+        return $entry;
+    }
+}
+
+sub restrict {
+    my ($self, $input_file, $roi_file, $output_file) = @_;
+    my $bedtools = $self->bedtools;
+    $DB::single = 1;
+    my @args = ($bedtools, "intersect", "-header", "-a", $input_file, "-b", $roi_file, "> $output_file");
+    Genome::Sys->shellcmd( cmd => join(" ", @args) );
```

Comment:
I'm going to keep this around.  The `_run` wrapper was initially introduced by @tabbott .  It's been useful when debugging various commands.  It serves as a centralized point to debug commands right before they are executed.